cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/SortedSSTableWriter.java

```diff
@@ -97,13 +111,26 @@ public String getPackageVersion(String lowestCassandraVersion)
         return CASSANDRA_VERSION_PREFIX + lowestCassandraVersion;
     }
 
+    /**
+     * Add a row to be written.
+     * @param token the hashed token of the row's partition key.
+     *              The value must be monotonically increasing in the subsequent calls.
+     * @param boundValues bound values of the columns in the row
+     * @throws IOException I/O exception when adding the row
+     */
     public void addRow(BigInteger token, Map<String, Object> boundValues) throws IOException
     {
-        if (minToken == null)
+        // first row
+        if (rowCount == 0)
         {
             minToken = token;
+            maxToken = token;
+        }
+        else
+        {
+            // rows are sorted. Therefore, only update the maxToken
+            maxToken = token;
```

Comment:
NIT: maxToken is always updated so no need to have it in an if/else branch
```suggestion
        if (rowCount == 0)
        {
            minToken = token;
        }
        maxToken = token;
```